lib/Dialect/HW/HWAttributes.cpp

```diff
@@ -210,14 +210,30 @@ EnumFieldAttr EnumFieldAttr::get(Location loc, StringAttr value,
 // InnerRefAttr
 //===----------------------------------------------------------------------===//
 
-Attribute InnerRefAttr::parse(AsmParser &p, Type type) {
+Attribute circt::hw::parseInnerRefAttr(AsmParser &p, bool parseLess) {
```

Comment:
The attribute should have a unique parse.  Is there a reason to have `<` and `>` at all?

Reply:
> Is there a reason to have < and > at all?

Just following the semantics of the existing `InnerRefAttr` parsing logic. This has been extracted out purely to avoid code duplication in `ibis.scoperef` **type** which wants to parse a `hw.innerref` as `!ibis.scoperef<@a::@b>`.